src/executeCommandLine/executeCommandLine.ts

```diff
@@ -4,12 +4,57 @@ namespace ts {
         value: string;
     }
 
-    function countLines(program: Program): number {
-        let count = 0;
+    function countLines(program: Program): Map<number> {
+        const counts = getCountsMap();
         forEach(program.getSourceFiles(), file => {
-            count += getLineStarts(file).length;
+            const key = getCountKey(program, file);
+            const lineCount = getLineStarts(file).length;
+            counts.set(key, counts.get(key)! + lineCount);
         });
-        return count;
+        return counts;
+    }
+
+    function countNodes(program: Program): Map<number> {
+        const counts = getCountsMap();
+        forEach(program.getSourceFiles(), file => {
+            const key = getCountKey(program, file);
+            counts.set(key, counts.get(key)! + file.nodeCount);
+        });
+        return counts;
+    }
+
+    function getCountsMap() {
+        const counts = createMap<number>();
+        counts.set("Library", 0);
+        counts.set("Definitions", 0);
+        counts.set("TypeScript", 0);
+        counts.set("JavaScript", 0);
+        counts.set("JSON", 0);
+        counts.set("Other", 0);
+        return counts;
+    }
+
+    function getCountKey(program: Program, file: SourceFile) {
+        if (program.isSourceFileDefaultLibrary(file)) {
+            return "Library";
+        }
+
+        const path = file.path;
+        if (fileExtensionIs(path, Extension.Dts)) {

```

Comment:
May be external library import as  opposed to your own d.ts is also useful ?

Reply:
What would be an example of consuming your own .d.ts?  Maybe you mean from referenced projects?

Reply:
yeah

Reply:
I haven't found that I've needed that in my own investigations but, to be fair, I don't often investigate composite projects.  Is there a scenario you have in mind?  Also, how would I distinguish those files?

Reply:
I could split out paths containing `node_modules`, but I'm concerned that would produce misleading/confusing results for mono-repos.  Unless you feel strongly, I'd like to go ahead with this as a first step.

Reply:
Sure but i think you should actually check isDeclarationFile instead of file.path ? to make sure project reference redirects with d.ts actually get counted as d.ts vs ts files since path will be input file Path if thats what it resolved to and resolvedPath would be d.ts path. Look for more info on path and resolvedPath in their jsdoc.

Reply:
Cool - I didn't know about that.